python/lsst/ts/ATHexapod/controller.py

```diff
@@ -0,0 +1,502 @@
+        """Generate qieryForPositionChange command string.
+
+        Queries wheter the axis positions have changed since the last position query was sent.
+        Response:
+        The response <uint> is bit-mappet and returned as the hexadecimal sum of the following codes:
+        1 = Position of the first axis has changed
+        2 = Position of the second axis has changed
+        4 = Posiiton of the third axis has changed
+        ...
+
+        Return
+        ------
+        pos_changed : tuple(bool, bool, bool, bool, bool, bool)
+
+        """
+        ret = await self.write_command("\6", multi_line=False)
+
+        code = int(ret.decode())
+
+        return tuple([(code & (1 << i)) > 0 for i in range(3)])
+
+    async def controller_ready(self):
+        """Generate requestControllerReadyStatus command string.
+
+        (p. 141) Request Controller Ready Status Asks controller for ready
+        status (tests if controller is ready to perform a new command)
+
+        B1h (ASCII character 177) if controller is ready B0h (ASCII character
+        176) if controller is not ready (e.g., performing a reference move)
+        """
+        ret = await self.write_command("\7", multi_line=False)
+
+        comp = ret == chr(177)
+        self.log.debug(f"{ret} : {chr(177)} : {comp}")
+
+        return ret == chr(177)
+
+    async def stop_all_axes(self):
+        """Generate stopAllAxes command string.
+
+        (p. 143) Stop All Axes To confirm that this worked, #5 has to be used.
+        """
+        return await self.write_command(chr(24), has_response=False)
+
+    async def set_position(self, x=None, y=None, z=None,
+                           u=None, v=None, w=None):
+        """Generate setTargetPosition command string.
+
+        (p. 206) Set Target Position
+
+        :Execute:
+        :Send: MOV X 10 U 5
+        :Note: Axis X moves to 10 (target
+        position in mm), axis U moves to
+        5 (target position in deg)
+
+        Parameters
+        ----------
+        x : `float`
+            X-axis position (in mm).
+        y : `float`
+            Y-axis position (in mm).
+        z : `float`
+            Z-axis position (in mm).
+        u : `float`
+            U-axis rotation (in deg).
+        v : `float`
+            V-axis rotation (in deg).
+        w : `float`
+            W-axis rotation (in deg).
+        """
+        target = ""
+        target += " X " + str(float(x)) if x is not None else ""
+        target += " Y " + str(float(y)) if y is not None else ""
+        target += " Z " + str(float(z)) if z is not None else ""
+        target += " U " + str(float(u)) if u is not None else ""
+        target += " V " + str(float(v)) if v is not None else ""
+        target += " W " + str(float(w)) if w is not None else ""
+
+        await self.write_command("MOV" + target, has_response=False)
+
+    async def referencing_result(self):
+        """Generate getReferencingResult command string.
```

Comment:
Your description implies that it returns the command string, not the results of sending the command.

I *strongly* suggest you change the brief description to something more like `Get referencing result.` (whatever that means) or`Execute the Get Referencing Result command and return the result.` or...?

Similarly for all your other commands described as "Generate x command string."